File: 113 - Path Sum II/main.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<IList<int>> PathSum(TreeNode root, int targetSum) {
        if(root == null){
            return [];
        }
        IList<IList<int>> valid_paths = new List<IList<int>>();

        paths(root, targetSum, 0, valid_paths, []);


        return valid_paths;

    }
    public void paths(TreeNode node, int targetSum, int cur_sum, IList<IList<int>> valid_paths, List<int> cur_path){
        cur_sum += node.val;
        cur_path.Add(node.val);
        if(node.right == null && node.left == null){
            if (cur_sum == targetSum){
                valid_paths.Add(new List<int>(cur_path));
            }
        }
        if(node.left != null){
            paths(node.left, targetSum, cur_sum, valid_paths, cur_path);
        }
       
        if(node.right != null){ 
            paths(node.right, targetSum, cur_sum, valid_paths, cur_path);
        }
        cur_path.RemoveAt(cur_path.Count - 1);
    }
}

File: 113 - Path Sum II/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def pathSum(self, root: Optional[TreeNode], targetSum: int) -> List[List[int]]:
        def get_sums(node, curSum, target, allSums):
            if node == None:
                return
            curSum.append(node.val)
            temp = list(curSum)
            if sum(curSum) == target and node.right == None and node.left == None:
                allSums.append(curSum)
            get_sums(node.right, curSum, target, allSums)
            get_sums(node.left, temp, target, allSums)
            
        allSums = []
        get_sums(root, [], targetSum, allSums)
        
        return allSums